rbac_api.go

```diff
@@ -203,6 +203,37 @@ func (e *Enforcer) GetImplicitRolesForUser(name string, domain ...string) ([]str
 	return res, nil
 }
 
+// GetImplicitUsersForRole gets implicit users for a role.
+func (e *Enforcer) GetImplicitUsersForRole(name string, domain ...string) ([]string, error) {
+	res := []string{}
+	roleSet := make(map[string]bool)
+	roleSet[name] = true
+
+	q := make([]string, 0)
+	q = append(q, name)
+
+	for len(q) > 0 {
+		name := q[0]
+		q = q[1:]
+
+		for _, rm := range e.rmMap {
+			roles, err := rm.GetUsers(name, domain...)
+			if err != nil && err.Error() != "error: name does not exist" {
+				return nil, err
+			}
+			for _, r := range roles {
+				if _, ok := roleSet[r]; !ok {
+					res = append(res, r)
+					q = append(q, r)
+					roleSet[r] = true
+				}
+			}
+		}
+	}
```

Comment:
I'm unfamiliar with go, so I'm probably reading it wrong, but does this mean that the following would be true?
```
g, alice, subgroup1
g2, subgroup1, topgroupA
```
results in GetImplicitUsersForRole(topgroupA) = [subgroup1, alice] ? And if so, would this not invalidate group isolation?

Reply:
@lanmarti @hsluoyz That a egacy issue.